examples/nextjs/pages/products/[handle].js

```diff
@@ -15,6 +15,11 @@ const ProductDetail = ({ product }) => {
     return <div>Loading...</div>;
   }
 
+  const { dispatchEvent } = useContext(EventLogContext);
```

Comment:
1. `EventLogContext` is `undefined` and needs to be imported
2. `useContext` must go above the `if` block, since hooks [can't be called conditionally](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level).